pkg/log/log.go

```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 The SQLFlow Authors. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/natefinch/lumberjack.v2"
+)
+
+// Factory builds Logger
+type Factory struct{}
+
+// Logger wraps logrus.Entry
+type Logger struct {
+	*logrus.Entry
+}
+
+// New sets logrus'output with lumberjack
+// filename: /path/to/log, e.g. "/var/log/sqlflow.log"
+func New(filename string) *Factory {
+	if len(strings.Trim(filename, " ")) > 0 {
+		logrus.SetOutput(&lumberjack.Logger{
```

Comment:
`logrus.SetOutput` sets the output of the global logger. https://github.com/sirupsen/logrus/blob/7ea96a3284ed9d25de5056804d60426cb75833c7/exported.go#L18-L20

However, `New` usually creates a new instance.